src/wp-admin/includes/image.php

```diff
@@ -261,142 +261,138 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
-		 *
-		 *     @type int $0 The image width.
-		 *     @type int $1 The image height.
-		 * }
-		 * @param string $file          Full path to the uploaded image file.
-		 * @param int    $attachment_id Attachment post ID.
-		 */
-		$threshold = (int) apply_filters( 'big_image_size_threshold', 2560, $imagesize, $file, $attachment_id );
+	/*
+		* If the original image's dimensions are over the threshold,
+		* scale the image and use it as the "full" size.
+		*/
+	$scale_down = false;
+	$convert    = false;
+
+	if ( $threshold && ( $image_meta['width'] > $threshold || $image_meta['height'] > $threshold ) ) {
+		// The image will be converted if needed on saving.
+		$scale_down = true;
+	} else {
+		// The image may need to be converted regardless of its dimensions.
+		$output_format = wp_get_image_editor_output_format( $file, $imagesize['mime'] );
 
-		/*
-		 * If the original image's dimensions are over the threshold,
-		 * scale the image and use it as the "full" size.
-		 */
-		$scale_down = false;
-		$convert    = false;
+		if (
+			is_array( $output_format ) &&
+			array_key_exists( $imagesize['mime'], $output_format ) &&
+			$output_format[ $imagesize['mime'] ] !== $imagesize['mime']
+		) {
+			$convert = true;
+		}
+	}
 
-		if ( $threshold && ( $image_meta['width'] > $threshold || $image_meta['height'] > $threshold ) ) {
-			// The image will be converted if needed on saving.
-			$scale_down = true;
-		} else {
-			// The image may need to be converted regardless of its dimensions.
-			$output_format = wp_get_image_editor_output_format( $file, $imagesize['mime'] );
+	if ( $scale_down || $convert ) {
+		$editor = wp_get_image_editor( $file );
 
-			if (
-				is_array( $output_format ) &&
-				array_key_exists( $imagesize['mime'], $output_format ) &&
-				$output_format[ $imagesize['mime'] ] !== $imagesize['mime']
-			) {
-				$convert = true;
-			}
+		if ( is_wp_error( $editor ) ) {
+			// This image cannot be edited.
+			return $image_meta;
 		}
 
-		if ( $scale_down || $convert ) {
-			$editor = wp_get_image_editor( $file );
+		if ( $scale_down ) {
+			// Resize the image. This will also convet it if needed.
+			$resized = $editor->resize( $threshold, $threshold );
+		} elseif ( $convert ) {
+			// The image will be converted (if possible) when saved.
+			$resized = true;
+		}
 
-			if ( is_wp_error( $editor ) ) {
-				// This image cannot be edited.
-				return $image_meta;
-			}
+		$rotated = null;
+
+		// If there is EXIF data, rotate according to EXIF Orientation.
+		if ( ! is_wp_error( $resized ) && is_array( $exif_meta ) ) {
+			$resized = $editor->maybe_exif_rotate();
+			$rotated = $resized; // bool true or WP_Error
+		}
 
+		if ( ! is_wp_error( $resized ) ) {
+			/*
+				* Append "-scaled" to the image file name. It will look like "my_image-scaled.jpg".
+				* This doesn't affect the sub-sizes names as they are generated from the original image (for best quality).
+				*/
```

Comment:
```suggestion
			/*
			 * Append "-scaled" to the image file name. It will look like "my_image-scaled.jpg".
			 * This doesn't affect the sub-sizes names as they are generated from the original image (for best quality).
			 */
```